catalog/async/catalog-async-postingestplugin/src/test/groovy/org/codice/ddf/catalog/async/processingplugin/ProcessingPostIngestPluginTest.groovy

```diff
@@ -0,0 +1,410 @@
+        when:
+        new ProcessingPostIngestPlugin(catalogFramework, processingFramework)
+
+        then:
+        thrown(NullPointerException)
+
+        where:
+        catalogFramework       | processingFramework
+        null                   | Mock(ProcessingFramework)
+        Mock(CatalogFramework) | null
+        null                   | null
+    }
+
+    /*
+    test process(CreateResponse)
+     */
+
+    def 'test process CreateResponse'() {
+        given:
+        final String id = _
+        final String sourceId = _
+        def metacard = Mock(Metacard) {
+            getId() >> id
+            getSourceId() >> sourceId
+        }
+
+        def responseProperties = [a: _]
+
+        def input = Mock(CreateResponse) {
+            getRequest() >> Mock(CreateRequest) {
+                getProperties() >> properties
+            }
+
+            getCreatedMetacards() >> [metacard]
+
+            getProperties() >> responseProperties
+        }
+
+        when:
+        def output = processingPostIngestPlugin.process(input)
+
+        then:
+        output == input
+
+        1 * catalogFramework.getResource(_ as ResourceRequest, sourceId) >> Mock(ResourceResponse) {
+            getResource() >> Mock(Resource)
+        }
+        1 * processingFramework.submitCreate(_ as ProcessRequest<ProcessCreateItem>) >> { ProcessRequest<ProcessCreateItem> processCreateRequest ->
+            assert processCreateRequest.getProcessItems().size() == 1
+            final processCreateItem = processCreateRequest.getProcessItems().get(0)
+            assert processCreateItem.getMetacard() == metacard
+            assert postProcessCompleteEntryAdded(processCreateRequest.getProperties(), responseProperties)
+        }
+
+        where:
+        properties << [createFalseProcessingCompleteProperties(), createNotBooleanProcessingCompleteProperties(), [:]]
+    }
+
+    def 'test process CreateResponse with no created metacards'() {
+        given:
+        def responseProperties = [a: _]
+
+        def input = Mock(CreateResponse) {
+            getRequest() >> Mock(CreateRequest) {
+                getProperties() >> properties
+            }
+
+            getCreatedMetacards() >> []
+
+            getProperties() >> responseProperties
+        }
+
+        when:
+        def output = processingPostIngestPlugin.process(input)
+
+        then:
+        output == input
+
+        0 * catalogFramework._
+        1 * processingFramework.submitCreate(_ as ProcessRequest<ProcessCreateItem>) >> { ProcessRequest<ProcessCreateItem> processCreateRequest ->
+            assert processCreateRequest.getProcessItems().isEmpty()
+            assert postProcessCompleteEntryAdded(processCreateRequest.getProperties(), responseProperties)
+        }
+
+        where:
+        properties << [createFalseProcessingCompleteProperties(), createNotBooleanProcessingCompleteProperties(), [:]]
+    }
+
+    def 'test no process CreateResponse'() {
```

Comment:
✏️  Any way we could improve the description to capture the fact that this is testing the cases where input is `null` and the list of metacards is `null`? With the current description, we almost have to look at the test code to know what's being tested.